### PR TITLE
chore(eslint): Bump to latest eslint-sentry config

### PR DESCRIPTION
**Pull request by evanpurkhiser**
No real important fixes here, but needed to bump things for getsentry, so this keeps sentry in line.

This did adjust some things minorly:

 - Use ExternalLink when possible
 - (Mostly) better proptype detection

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/interfaces/exceptionMechanism.jsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ExceptionMechanism extends React.Component {
         }),
         signal: PropTypes.shape({
           number: PropTypes.number.isRequired,
-          code: PropTypes.nubmer,
+          code: PropTypes.number,
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/events/interfaces/exceptionMechanism.jsx]: 👍

File: src/sentry/static/sentry/app/components/events/contexts/user.jsx [hunk 2]
```diff
@@ -27,9 +28,9 @@ class UserContextType extends React.Component {
         <pre>
           {user.email}
           {EMAIL_REGEX.test(user.email) && (
-            <a href={`mailto:${user.email}`} target="_blank" className="external-icon">
+            <ExternalLink href={`mailto:${user.email}`} className="external-icon">
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/components/events/contexts/user.jsx]: Why doesn't it like `<a>` tag anymore?

File: src/sentry/static/sentry/app/views/settings/components/settingsNavigationGroup.jsx [hunk 3]
```diff
@@ -23,9 +23,11 @@ export default class NavigationGroup extends React.Component {
     ...SentryTypes.NavigationGroup,
     organization: SentryTypes.Organization,
     project: SentryTypes.Project,
-    access: PropTypes.object,
-    features: PropTypes.object,
-    id: PropTypes.string,
+
+    // Used in the `show` and `badge` function
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/components/settingsNavigationGroup.jsx]: Thanks for the comment

File: src/sentry/static/sentry/app/views/settings/components/settingsPluginsNavigation.jsx [hunk 4]
```diff
@@ -22,8 +22,9 @@ class SettingsPluginsNavigation extends React.Component {
     ...SentryTypes.NavigationGroup,
     organization: SentryTypes.Organization,
     project: SentryTypes.Project,
-    access: PropTypes.object,
-    features: PropTypes.object,
+
+    access: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
+    features: PropTypes.object, // eslint-disable-line react/no-unused-prop-types
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/components/settingsPluginsNavigation.jsx]: Both `access` and `features` don't appear to be used? Why do we need them?

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/settingsPluginsNavigation.jsx]: Ah you're right, I thought this was the same thing as the other one. Yeah I'll remove.

File: src/sentry/static/sentry/app/components/events/contexts/user.jsx (hunk 2 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/events/contexts/user.jsx]: `target="_blank"` without `noopener`

File: src/sentry/static/sentry/app/views/settings/components/settingsPluginsNavigation.jsx (hunk 4 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/settingsPluginsNavigation.jsx]: Fixed.

**Review by lynnagara — APPROVED** [review]

**Pull request merged**